rust/src/lib.rs

```diff
@@ -125,13 +124,46 @@ fn block_db(fsblock_db: *const u8, fsblock_db_len: usize) -> anyhow::Result<FsBl
         .map_err(|e| anyhow!("Error opening block source database connection: {}", e))
 }
 
-fn account_id_from_i32(account: i32) -> anyhow::Result<AccountId> {
+fn account_id_from_i32(account: i32) -> anyhow::Result<zip32::AccountId> {
     u32::try_from(account)
         .map_err(|_| ())
-        .and_then(|id| AccountId::try_from(id).map_err(|_| ()))
+        .and_then(|id| zip32::AccountId::try_from(id).map_err(|_| ()))
         .map_err(|_| anyhow!("Invalid account ID"))
 }
 
+fn account_id_from_ffi<P: Parameters>(
+    db_data: &WalletDb<rusqlite::Connection, P>,
+    account_index: i32,
+) -> anyhow::Result<AccountId> {
+    let requested_account_index = account_id_from_i32(account_index)?;
+
+    // Find the single account matching the given ZIP 32 account index.
+    let mut accounts = db_data
+        .get_account_ids()?
+        .into_iter()
+        .filter_map(|account_id| {
```

Comment:
Wouldn't `find_map` be better here?

Reply:
No, I want it to be an error if there is more than one matching ZIP 32 account index.

Reply:
Ah, okay, I glossed over that `accounts.next()` was being called twice.